plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/IngestModule.java

```diff
@@ -29,6 +32,10 @@
     protected void configure() {
         binder().bind(IngestRestFilter.class).asEagerSingleton();
         binder().bind(PipelineStore.class).asEagerSingleton();
+        binder().bind(PipelineConfigDocReader.class).asEagerSingleton();
+        Multibinder<Processor.Builder.Factory> multibinder
```

Comment:
I would keep a `Map<String, Classs<? extends Processor.Builder.Factory>>` as a variable on the module... add a `public void registerProcessor(String, Classs<? extends Processor.Builder.Factory>)` method to the module and change this multibinder to a map binder that will just map all the registered processors.